packages/react-button/src/components/CompoundButton/useCompoundButtonStyles.ts

```diff
@@ -0,0 +1,271 @@
+
+    return {
+      fontSize: compoundButtonTokens.small.fontSize,
+      lineHeight: compoundButtonTokens.small.lineHeight,
+    };
+  },
+  icon: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      fontSize: compoundButtonTokens.base.iconFontSize,
+      height: compoundButtonTokens.base.iconHeight,
+      width: compoundButtonTokens.base.iconWidth,
+    };
+  },
+  contentContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    textAlign: 'left',
+  },
+  secondaryContent: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      lineHeight: '100%',
+
+      color: compoundButtonTokens.base.secondaryContentColor,
+      fontSize: compoundButtonTokens.base.secondaryContentFontSize,
+      fontWeight: compoundButtonTokens.base.secondaryContentFontWeight,
+      marginTop: compoundButtonTokens.base.secondaryContentGap,
+
+      ':hover': {
+        color:
+          compoundButtonTokens.base.hovered?.secondaryContentColor || compoundButtonTokens.base.secondaryContentColor,
+      },
+
+      ':active': {
+        color:
+          compoundButtonTokens.base.pressed?.secondaryContentColor ||
+          compoundButtonTokens.base.hovered?.secondaryContentColor ||
+          compoundButtonTokens.base.secondaryContentColor,
+      },
+    };
+  },
+  secondaryContentLarge: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      fontSize: compoundButtonTokens.large.secondaryContentFontSize,
+    };
+  },
+  secondaryContentPrimary: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      color: compoundButtonTokens.primary.secondaryContentColor,
+
+      ':hover': {
+        color: compoundButtonTokens.primary.hovered?.secondaryContentColor,
+      },
+
+      ':active': {
+        color: compoundButtonTokens.primary.pressed?.secondaryContentColor,
+      },
+    };
+  },
+  secondaryContentDisabled: theme => {
+    const compoundButtonTokens = makeCompoundButtonTokens(theme);
+
+    return {
+      color: compoundButtonTokens.disabled.secondaryContentColor,
+
+      ':hover': {
+        color: compoundButtonTokens.disabled.hovered?.secondaryContentColor,
+      },
+
+      ':active': {
+        color: compoundButtonTokens.disabled.pressed?.secondaryContentColor,
+      },
+    };
+  },
+});
+
+export const useCompoundButtonStyles = (state: CompoundButtonState, selectors: CompoundButtonStyleSelectors) => {
+  // Save the class names used in useButtonStyles and undefine them at the state level so that they are always applied
+  // last.
+  const {
+    className: rootClassName,
+    children: { className: childrenClassName } = { className: undefined },
+    icon: { className: iconClassName } = { className: undefined },
+  } = state;
+  state.className = undefined;
+  if (state.children) {
+    state.children.className = undefined;
```

Comment:
What is `state.children`? is this guaranteed to be a props object, or can it be jsx or a string?

Reply:
It is treated as a slot so it can be any of those.

Reply:
In here it is guaranteed to be an object after going through `useCompoundButton` though.